### PR TITLE
Chore: deployment scripts

### PR DESCRIPTION
**Pull request by haythemsellami**
# Task: deployment scripts

**Should be reviewed/merged after #954**

## Description

Instructions how to deploy in the README.

Fixes ENG-1373

## Type of change

- [X] New feature
- [ ] Bug fix
- [ ] Testing code
- [ ] Document update or config files

### CODE REVIEW

**Review by alpinechicken — APPROVED** [review]
lgtm

File: packages/zen-bull-netting/src/NettingLib.sol [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+        } else {
+            IERC20(oSqth).transfer(trader, oSqthBalance);
+
+            emit DepositAuction(trader, bidId, oSqthBalance);
+
+            return (true, 0);
+        }
+    }
+
+    function transferWethToMarketMaker(
```
Comment (aleone) [inline packages/zen-bull-netting/src/NettingLib.sol:72]: why did you do this one differently from the other weth one?

File: packages/zen-bull-netting/src/ZenBullNetting.sol [hunk 2]
```diff
@@ -573,96 +549,73 @@ contract ZenBullNetting is Ownable, EIP712, FlashSwap {
...
-            usdcToBorrow = bullToMint * IEulerSimpleLens(eulerLens).getDTokenBalance(usdc, zenBull)
-                / bullTotalSupply;
-        }
-
-        // deposit into crab
-        uint256 wethFromAuction = IWETH(weth).balanceOf(address(this));
-        IWETH(weth).withdraw(wethFromAuction);
-
-        ICrabStrategyV2(crab).deposit{value: ethIntoCrab}();
+            (uint256 wethToLend, uint256 usdcToBorrow) = NettingLib.caclWethToLendAndUsdcToBorrow(
```
Comment (aleone) [inline packages/zen-bull-netting/src/ZenBullNetting.sol:565]: calcWethToLendAndUsdcToBorrow vs cacl

File: packages/zen-bull-netting/src/ZenBullNetting.sol [hunk 3]
```diff
@@ -695,11 +648,7 @@ contract ZenBullNetting is Ownable, EIP712, FlashSwap {
 
                     wethAmountToSend =
                         depositReceipt.amount * memVar.remainingEth / _params.depositsToProcess;
-                    if (wethAmountToSend > 1e12) {
-                        payable(depositReceipt.sender).sendValue(wethAmountToSend);
-                    } else {
-                        wethAmountToSend = 0;
-                    }
+                    payable(depositReceipt.sender).sendValue(wethAmountToSend);
```
Comment (aleone) [inline packages/zen-bull-netting/src/ZenBullNetting.sol:651]: did we have to remove this due to bytecode size?

File: packages/zen-bull-netting/src/ZenBullNetting.sol [hunk 4]
```diff
@@ -787,32 +723,23 @@ contract ZenBullNetting is Ownable, EIP712, FlashSwap {
...
-                oSqthQuantity = toExchange;
-            }
-
-            IERC20(weth).transfer(
-                _params.orders[i].trader, (oSqthQuantity * _params.clearingPrice) / 1e18
-            );
-            toExchange -= oSqthQuantity;
-
-            emit WithdrawAuction(
+            toExchange = NettingLib.transferWethToMarketMaker(
```
Comment (aleone) [inline packages/zen-bull-netting/src/ZenBullNetting.sol:735]: I don't think this function works.  ```        uint256 oSqthQuantity;``` is not used

Doesn't it always send 0 for oSQTHQuantity (which is supposed to be the MM order size)?

File: packages/zen-bull-netting/src/ZenBullNetting.sol [hunk 5]
```diff
@@ -862,28 +789,26 @@ contract ZenBullNetting is Ownable, EIP712, FlashSwap {
...
+            IZenBullStrategy(zenBull).withdraw(zenBullAmountToBurn);
+            IWETH(weth).deposit{value: amountToPay}();
+            IWETH(weth).transfer(pool, amountToPay);
+        } else if (callSource == 1) {
+            uint256 wethToLend = abi.decode(callData, (uint256));
 
             IWETH(weth).withdraw(IWETH(weth).balanceOf(address(this)));
-            IZenBullStrategy(zenBull).deposit{value: data.wethToLend}(
+            IZenBullStrategy(zenBull).deposit{value: wethToLend}(
                 IERC20(crab).balanceOf(address(this))
```
Comment (aleone) [inline packages/zen-bull-netting/src/ZenBullNetting.sol:809]: is it possible you get a state here where we have more crab then expected, which means that we deposit too much crab.  Lets say someone sends 1 wei to the contract and then you try to deposit with 1 wei more than you sent as crab param, that means you won't have enough eth sent I think?

**Review by aleone — CHANGES_REQUESTED** [review]
some comments, one MM transfer function I think doesnt work correctly

File: packages/zen-bull-netting/src/NettingLib.sol [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+    event DepositAuction(address indexed trader, uint256 indexed bidId, uint256 quantity);
+    event WithdrawAuction(
+        address indexed trader, uint256 indexed bidId, uint256 quantity, uint256 price
+    );
+
+    function transferWethFromMarketMakers(
+        address weth,
+        address trader,
+        uint256 quantity,
+        uint256 oSqthToMint,
```
[the remark this answers is not in the document]
Comment (haythemsellami) [reply packages/zen-bull-netting/src/NettingLib.sol:21]: SO initially that function take the oSQTH that will be minted, after the first transfer from MM it return the remaining, and then it is passed again as arg.

File: packages/zen-bull-netting/src/NettingLib.sol [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+        uint256 clearingPrice
+    ) external returns (bool, uint256) {
+        if (quantity >= oSqthToMint) {
+            IWETH(weth).transferFrom(trader, address(this), (oSqthToMint * clearingPrice) / 1e18);
```
[the remark this answers is not in the document]
Comment (haythemsellami) [reply packages/zen-bull-netting/src/NettingLib.sol:25]: Will fix events in different PR

File: packages/zen-bull-netting/src/ZenBullNetting.sol (hunk 3 above)
Comment (haythemsellami) [reply packages/zen-bull-netting/src/ZenBullNetting.sol:651]: yeah, but can be added now as there is enough space

File: packages/zen-bull-netting/src/ZenBullNetting.sol (hunk 4 above)
Comment (haythemsellami) [reply packages/zen-bull-netting/src/ZenBullNetting.sol:734]: fixed and added assert in test

File: packages/zen-bull-netting/src/ZenBullNetting.sol (hunk 2 above)
Comment (haythemsellami) [reply packages/zen-bull-netting/src/ZenBullNetting.sol]: fixed

File: packages/zen-bull-netting/src/NettingLib.sol (hunk 1 above)
Comment (haythemsellami) [reply packages/zen-bull-netting/src/NettingLib.sol:72]: which one ?

File: packages/zen-bull-netting/src/ZenBullNetting.sol [hunk 8]
```diff
@@ -248,6 +223,11 @@ contract ZenBullNetting is Ownable, EIP712, FlashSwap {
         uint256 ethAmount,
         uint256 indexed receiptIndex
     );
+    /// @dev shared events with the NettingLib for client side to detect them
+    event DepositAuction(address indexed trader, uint256 indexed bidId, uint256 quantity);
```
[the remark this answers is not in the document]
Comment (haythemsellami) [reply packages/zen-bull-netting/src/ZenBullNetting.sol:227]: Will fix events in different PR

File: packages/zen-bull-netting/src/NettingLib.sol [hunk 9]
```diff
@@ -0,0 +1,180 @@
...
+            IWETH(weth).transferFrom(trader, address(this), (quantity * clearingPrice) / 1e18);
+
+            return (false, (oSqthToMint - quantity));
+        }
+    }
+
+    function transferOsqthFromMarketMakers(
+        address oSqth,
+        address trader,
+        uint256 toExchange,
```
[the remark this answers is not in the document]
Comment (haythemsellami) [reply packages/zen-bull-netting/src/NettingLib.sol]: fixed

File: packages/zen-bull-netting/src/ZenBullNetting.sol (hunk 5 above)
Comment (haythemsellami) [reply packages/zen-bull-netting/src/ZenBullNetting.sol:808]: I don't think so...

**Pull request merged**